File: Greenhouse-backend/src/main/java/dev/canercin/greenhouseiot/dto/GreenhouseMonthlyAverageDTO.java
package dev.canercin.greenhouseiot.dto;

import lombok.Data;
import java.time.YearMonth;

@Data
public class GreenhouseMonthlyAverageDTO {
    private YearMonth yearMonth;
    private double avgHumidity;
    private double avgLight;
    private double avgSoilMoisture;
    private double avgTemperatureAsc;
    private double avgTemperatureAsf;
    private double avgWaterLevel;
}

File: Greenhouse-FrontEnd/src/app/models/GreenhouseMonthlyAverageDTO.ts
export class GreenhouseMonthlyAverageDTO {
  yearMonth: string;
  avgHumidity: number;
  avgLight: number;
  avgSoilMoisture: number;
  avgTemperatureAsc: number;
  avgTemperatureAsf: number;
  avgWaterLevel: number;

  constructor(yearMonth: string, avgHumidity: number, avgLight: number, avgSoilMoisture: number, avgTemperatureAsc: number, avgTemperatureAsf: number, avgWaterLevel: number) {
    this.yearMonth = yearMonth;
    this.avgHumidity = avgHumidity;
    this.avgLight = avgLight;
    this.avgSoilMoisture = avgSoilMoisture;
    this.avgTemperatureAsc = avgTemperatureAsc;
    this.avgTemperatureAsf = avgTemperatureAsf;
    this.avgWaterLevel = avgWaterLevel;
  }
}
